### PR TITLE
[Resteasy 3392] Migrate the server-adapters module to JUnit 5

### PR DESCRIPTION
**Pull request by Dkafetzis**
https://issues.redhat.com/browse/RESTEASY-3392

### CODE REVIEW

**Comment by jamezp** [comment]
This doesn't compile for me.

**Comment by Dkafetzis** [comment]
@jamezp Should be fine now

File: server-adapters/resteasy-netty4/pom.xml [hunk 1]
```diff
@@ -83,8 +83,13 @@
...
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
```
Comment (jamezp) [inline server-adapters/resteasy-netty4/pom.xml:92]: These should be:
```xml
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter</artifactId>
    <scope>test</scope>
</dependency>
```

File: server-adapters/server-adapter-test-base/pom.xml [hunk 2]
```diff
@@ -54,9 +54,12 @@
...
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <scope>compile</scope>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
```
Comment (jamezp) [inline server-adapters/server-adapter-test-base/pom.xml:62]: These should be:
```xml
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter</artifactId>
    <scope>test</scope>
</dependency>
```

File: server-adapters/resteasy-jdk-http/pom.xml [hunk 3]
```diff
@@ -45,8 +45,13 @@
...
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
```
Comment (jamezp) [inline server-adapters/resteasy-jdk-http/pom.xml:54]: These should be:
```xml
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter</artifactId>
    <scope>test</scope>
</dependency>
```

File: server-adapters/resteasy-netty4-cdi/pom.xml [hunk 4]
```diff
@@ -38,8 +38,8 @@
 
         <!-- Test dependencies -->
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
```
Comment (jamezp) [inline server-adapters/resteasy-netty4-cdi/pom.xml:42]: If this one does not work with JUnit 5 we should file a JIRA for this and make a comment as to why with the JIRA referenced.

File: server-adapters/resteasy-reactor-netty/pom.xml [hunk 5]
```diff
@@ -105,8 +105,13 @@
...
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
```
Comment (jamezp) [inline server-adapters/resteasy-reactor-netty/pom.xml:114]: These should be:
```xml
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter</artifactId>
    <scope>test</scope>
</dependency>
```

File: server-adapters/resteasy-undertow-cdi/pom.xml [hunk 6]
```diff
@@ -104,8 +104,13 @@
...
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
```
Comment (jamezp) [inline server-adapters/resteasy-undertow-cdi/pom.xml:113]: These should be:
```xml
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter</artifactId>
    <scope>test</scope>
</dependency>
```

File: server-adapters/resteasy-undertow/pom.xml [hunk 7]
```diff
@@ -39,8 +39,13 @@
...
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
```
Comment (jamezp) [inline server-adapters/resteasy-undertow/pom.xml:48]: These should be:
```xml
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter</artifactId>
    <scope>test</scope>
</dependency>
```

File: server-adapters/resteasy-vertx/pom.xml [hunk 8]
```diff
@@ -132,8 +132,13 @@
...
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
```
Comment (jamezp) [inline server-adapters/resteasy-vertx/pom.xml:141]: These should be:
```xml
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter</artifactId>
    <scope>test</scope>
</dependency>
```

**Comment by Dkafetzis** [comment]
@jamezp No problem. As for the issues, i just understood that it had to be a PR for each module, not for every submodule. Will change it on the following PRs

**Review by jamezp — APPROVED** [review]
I'll merge this, but I'm not sure the junit-jupiter-api is the correct dependency. Per the [documentation](https://junit.org/junit5/docs/current/user-guide/#running-tests-build-maven) the dependency should be `org.junit.jupiter:junit-jupiter`.

**Pull request merged**